pkg/athena/datasource.go

```diff
@@ -159,3 +160,20 @@ func (s *AthenaDatasource) Workgroups(ctx context.Context, options sqlds.Options
 	}
 	return api.Workgroups(ctx)
 }
+
+func parseArgs(queryArgs json.RawMessage) (sqlds.Options, error) {
+	args := map[string]interface{}{}
+	if queryArgs != nil {
+		err := json.Unmarshal(queryArgs, &args)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse query args: %w", err)
+		}
+	}
+
+	options := sqlds.Options{}
+	for k, v := range args {
+		options[k] = fmt.Sprintf("%v", v)
+	}
+
+	return options, nil
+}
```

Comment:
For the args like “database” etc in the map it might be nice to have this in constants which can also be used in the Apply function. It’ll be nice to know where these are used again. Let me know if this makes sense (writing on mobile sorry)